src/credentials/FabricTable.cpp

```diff
@@ -317,15 +317,25 @@ CHIP_ERROR FabricInfo::VerifyCredentials(const ByteSpan & noc, const ByteSpan &
     NodeId nodeId;
     ReturnErrorOnFailure(ExtractNodeIdFabricIdFromOpCert(certificates.GetLastCert()[0], &nodeId, &fabricId));
 
+    FabricId icacFabric = kUndefinedFabricId;
     if (!icac.empty())
     {
-        FabricId icacFabric = kUndefinedFabricId;
         if (ExtractFabricIdFromCert(certificates.GetCertSet()[1], &icacFabric) == CHIP_NO_ERROR && icacFabric != kUndefinedFabricId)
         {
             ReturnErrorCodeIf(icacFabric != fabricId, CHIP_ERROR_FABRIC_MISMATCH_ON_ICA);
         }
     }
 
+    FabricId rootFabric = kUndefinedFabricId;
+    if (ExtractFabricIdFromCert(certificates.GetCertSet()[0], &rootFabric) == CHIP_NO_ERROR && rootFabric != kUndefinedFabricId)
+    {
+        if (!icac.empty())
+        {
+            ReturnErrorCodeIf(rootFabric != icacFabric, CHIP_ERROR_FABRIC_MISMATCH_ON_ICA);
+        }
+        ReturnErrorCodeIf(rootFabric != fabricId, CHIP_ERROR_FABRIC_MISMATCH_ON_ICA);
```

Comment:
This is not a mismatch on the ICA...

Reply:
Replaced it with `CHIP_ERROR_WRONG_CERT_DN`. I don't know if introduction of a new error code is justified here for this case.